pkg/azure/testhelp/fakes/machineresources.go

```diff
@@ -151,6 +151,14 @@ func (m *MachineResources) UpdateDataDisksDeleteOpt(deleteOpt *armcompute.DiskDe
 	}
 }
 
+// AttachDataDisk attaches a data disk to the VM
+func (m *MachineResources) AttachDataDisk(spec api.AzureProviderSpec, diskName string, lun int32, deleteOption armcompute.DiskDeleteOptionTypes) {
```

Comment:
instead of taking a lun you can get the next lun value simply by inspecting the number of data disks that are already there. This will remove the need for the developer to provide a unique value. This method will be used to dynamically attach disks to the VM. So either you provide a convenience method to get the next unused lun or just to it within the implementation and not allow the consumer to provide one. Thus avoiding any mistakes and also making it easier.